### PR TITLE
Move docker compose services used for development into a separate compose file

### LINKED ISSUE #2788: Create a separate docker-compose file for production
## Problem
The docker-compose file contains development, demo, and testing services which can be [confusing](https://www.reddit.com/r/selfhosted/comments/11n2fxx/comment/jbnmdvi/?utm_source=share&utm_medium=web2x&context=3) to a user looking to just run the production version of Mathesar.

## Proposed solution
Create a separate docker-compose file that will contain only production services.

### PR DESCRIPTION
**Pull request by silentninja**
Fixes #2788 

- Moves the docker services used during testing and development(`test-service` used for running test cases and `dev-service` used when developing Mathesar) into `docker-compose.dev.yml`
- Removes docker profiles as they are irrelevant now due to the changes to docker compose files.
- Adds comments to all the docker compose services, so that it is easy to get an understanding of the services.
- Updates the docker compose commands used in the documentation to reflect the changes to the docker compose changes made by this PR.

**Technical details**

**Screenshots**

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: docs/docs/install/docker-compose/under-the-hood.md [hunk 1]
```diff
@@ -42,6 +42,7 @@ We need to store all the details configured above, and we do so in a file in you
 We'll store two files under that directory:
```
Comment (mathemancer) [inline docs/docs/install/docker-compose/under-the-hood.md:42]:
```suggestion
We'll store three files under that directory:
```

File: install.sh [hunk 2]
```diff
@@ -428,14 +428,16 @@ installation.
 "
 printf "Downloading docker-compose.yml...
```
Comment (mathemancer) [inline install.sh:429]: This needs to be updated to be accurate.

**Review by mathemancer — CHANGES_REQUESTED** [review]
I requested a few changes.

Overall, if the goal is to make the docker compose setup easier to understand and modify (e.g., punching in environment variables), I think this is a step backwards rather than forwards. It's now even more difficult to understand how environment variables are defined, and to which services they're applied, due to the separate `docker-compose.common.yml` file. This won't make sense in a 'prod' install, since there's only that one and the one you call.

Maybe this could be fixed with more comments in the 'main' (not common) docker compose file (e.g., in the service to explain that the inherited service has the environment variable plumbing), but I'm skeptical of this organizational setup overall.

**Comment by silentninja** [comment]
@mathemancer I went with `docker-compose.common.yml` because I wanted the keep the option of running the command `docker-compose up` possible without using `profiles`. I can remove the `base service`, making the `service` as the parent service `dev-service` extends from, and then address the possibility of running a Mathesar web server without creating a `db` in the documentation.

**Comment by silentninja** [comment]
@mathemancer I removed the `docker-compose.common.yml` as per our discussion over the sync call. Can you review it again?

**Pull request merged**